pkgpanda/http/__init__.py

```
@@ -0,0 +1,170 @@
+
+
+app = Flask(__name__)
+app.config.from_object('pkgpanda.http.config')
+app.config.from_envvar('PKGPANDA_HTTP_CONFIG', silent=True)
+
+
+@app.errorhandler(Exception)
+def uncaught_exception_handler(exc):
+    error_message = 'An unexpected error has occurred.'
+    logging.error(error_message, exc_info=exc)
+    return error_response(error_message), http.client.INTERNAL_SERVER_ERROR
+
+
+@app.before_request
+def set_app_attrs_from_config():
+    current_app.install = Install(
+        current_app.config['DCOS_ROOT'],
+        current_app.config['DCOS_CONFIG_DIR'],
+        current_app.config['DCOS_ROOTED_SYSTEMD'],
+        manage_systemd=True,
+        block_systemd=False)
+    current_app.repository = Repository(
+        current_app.config['DCOS_REPO_DIR'])
+
+
+@app.route('/', methods=['GET'])
+def get_package_list():
+    return package_listing_response(current_app.repository.list())
+
+
+@app.route('/<package_id>', methods=['GET'])
+def get_package(package_id):
+    if not current_app.repository.has_package(package_id):
+        return (
+            error_response('Package {} not found.'.format(package_id)),
+            http.client.NOT_FOUND,
+        )
+    return package_response(package_id, current_app.repository)
+
+
+@app.route('/<package_id>', methods=['POST'])
+def fetch_package(package_id):
+    try:
+        repository_url = request.json['repository_url']
+    except Exception:
+        return (
+            error_response(
+                'Request body must be a json object with a `repository_url` '
+                'key.'
+            ),
+            http.client.BAD_REQUEST,
+        )
+
+    actions.fetch_package(
+        current_app.repository,
+        repository_url,
+        package_id,
+        current_app.config['WORK_DIR'])
+
+    return empty_response
+
+
+@app.route('/<package_id>', methods=['DELETE'])
+def remove_package(package_id):
+    if not current_app.repository.has_package(package_id):
```

`remove_package()` silently succeeds if the package doesn't exist 😢 https://github.com/dcos/dcos/blob/db8b5678caf6a958b1926a50f65ce9a37adf96b6/pkgpanda/__init__.py#L387 I added this check because I didn't want to change CLI behavior as part of this PR. I guess I could add the exception, remove this check, and have the CLI catch and ignore it?